runtime/lua/vim/fs.lua

```diff
@@ -144,7 +144,7 @@ end
 --- The search can be narrowed to find only files or only directories by
 --- specifying {type} to be "file" or "directory", respectively.
 ---
----@param names (string|table|fun(name: string): boolean) Names of the files
+---@param names (string|table|fun(name: string, path: string): boolean) Names of the files
```

Comment:
Can you add a comment in the docstring about what `{path}` is. Maybe something like

```
If {names} is a function, it is called for each file and directory within the traversed directory and is passed the base name of each file or directory and the path of the directory being traversed. The function should return `true` if the given file or directory is considered a match.
```